doc/api/url.md

```diff
@@ -250,7 +250,308 @@ properties of URL objects:
 For example, the ASCII space character (`' '`) is encoded as `%20`. The ASCII
 forward slash (`/`) character is encoded as `%3C`.
 
+## The WHATWG URL API
+
+The `url` module provides an experimental implementation of the
+[WHATWG URL Standard][] as an alternative to the existing `url.parse()` API.
+
+```js
+const URL = require('url').URL;
+const myURL = new URL('https://example.org/foo');
+
+console.log(myURL.href);     // "https://example.org/foo"
+console.log(myURL.protocol); // "https:"
+console.log(myURL.hostname); // "/example.org"
+console.log(myURL.pathname); // "/foo"
+```
+
+### Class: URL
+#### Constructor: new URL(input[, base])
+
+* `input` {String} The input URL to parse
+* `base` {String | URL} The base URL to resolve against if the `input` is not
+  absolute.
+
+Creates a new `URL` object by parsing the `input` relative to the `base`. If
+`base` is passed as a string, it will be parsed equivalent to `new URL(base)`.
+
+```js
+const myURL = new URL('/foo', 'https://example.org/');
+  // https://example.org/foo
+```
+
+A `TypeError` will be thrown if the `input` or `base` are not valid URLs.
+
+#### url.hash
+
+Gets and sets the fragment portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/foo#bar');
+console.log(myURL.hash);
+  // Prints #bar
+
+myURL.hash = 'baz';
+console.log(myURL.href);
+  // Prints http://example.org/foo#baz
+```
+
+#### url.host
+
+Gets and sets the host portion of the URL.
+
+```js
+const myURL = new URL('http://example.org:81/foo');
+console.log(myURL.host);
+  // Prints example.org:81
+
+myURL.host = 'example.com:82';
+console.log(myURL.href);
+  // Prints http://example.com:82/foo
+```
+
+#### url.hostname
+
+Gets and sets the hostname portion of the URL. The key difference between
+`url.host` and `url.hostname` is that `url.hostname` does *not* include the
+port.
+
+```js
+const myURL = new URL('http://example.org:81/foo');
+console.log(myURL.hostname);
+  // Prints example.org
+
+myURL.hostname = 'example.com:82';
+console.log(myURL.href);
+  // Prints http://example.com:81/foo
+```
+
+#### url.href
+
+Gets and sets the serialized URL.
+
+```js
+const myURL = new URL('http://example.org/foo');
+console.log(myURL.href);
+  // Prints http://example.org/foo
+
+myURL.href = 'https://example.com/bar'
+  // Prints https://example.com/bar
+```
+
+#### url.origin
+
+Gets the read-only Unicode serialization of the URL's origin.
```

Comment:
That one is specificly pointed out in the spec https://url.spec.whatwg.org/#dom-url-origin

> The origin attribute’s getter must return the Unicode serialization of context object’s url’s origin. [HTML]
>
> It returns the Unicode rather than the ASCII serialization for compatibility with HTML’s MessageEvent feature. [HTML]